debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java

```diff
@@ -77,13 +74,13 @@ public long getNumberOfEmptyPolls() {
         return numberOfEmptyPolls.get();
     }
 
-    public void onSourceEventPolled(ChangeStreamDocument<BsonDocument> event, Clock clock, Instant prePollTimestamp) {
+    public void onSourceEventPolled(ResumableChangeStreamEvent<?> event, Clock clock, Instant prePollTimestamp) {
         var now = clock.currentTimeAsInstant();
 
         AtomicLong counter;
         AtomicLong timer;
 
-        if (event == null) {
+        if (event.isEmpty()) {
             counter = numberOfEmptyPolls;
```

Comment:
@jpechane  please remind me which of these metrics we wanted to keep? poll time for event / no event, count of no events? Count of events as added recently is duplicit IMHO and should be removed.

Reply:
Count of no events and poll time for event. IIRC poll time for no event did not provide any info

Reply:
OK... I will rearrange things to the right places in this PR and remove the useless metrics in a sepparrate one